plugins/InspectionGadgets/src/com/siyeh/ig/redundancy/RedundantStringOperationInspection.java

```diff
@@ -109,7 +112,7 @@ private static class RedundantStringOperationVisitor extends JavaElementVisitor
       .register(STRING_LAST_INDEX_OF, this::getLastIndexOfProblem)
       .register(STRING_IS_EMPTY, this::getRedundantCaseChangeProblem)
       .register(STRING_EQUALS, this::getRedundantSubstringEqualsProblem)
-      .register(anyOf(STRING_EQUALS, STRING_EQUALS_IGNORE_CASE), this::getRedundantCaseEqualsProblem)
```

Comment:
Do you really need to route to `getRedundantCaseEqualsProblem` by yourself? I guess, `CallMapper` would take care of this. You can register two handlers for the same CallMatcher.

Reply:
Done